dynamiq/nodes/tools/exa_search.py

```diff
@@ -0,0 +1,169 @@
+
+    group: Literal[NodeGroup.TOOLS] = NodeGroup.TOOLS
+    name: str = "Exa Search Tool"
+    description: str = (
+        "A tool for searching the web using Exa AI. "
+        "Provides advanced search capabilities with options for filtering results."
+    )
+    connection: Exa
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    input_schema: ClassVar[type[ExaInputSchema]] = ExaInputSchema
+
+    @staticmethod
+    def to_camel_case(snake_str: str) -> str:
+        """Convert snake_case to camelCase."""
+        components = snake_str.split("_")
+        return components[0] + "".join(x.title() for x in components[1:])
+
+    def _format_search_results(self, results: list[dict[str, Any]]) -> str:
+        """
+        Formats the search results into a human-readable string.
+
+        Args:
+            results (list[dict[str, Any]]): The raw search results.
+
+        Returns:
+            str: A formatted string containing the search results.
+        """
+        formatted_results = []
+        for result in results:
+            formatted_results.extend(
+                [
+                    f"Title: {result.get('title', 'N/A')}",
+                    f"URL: {result.get('url', 'N/A')}",
+                    f"Published Date: {result.get('publishedDate', 'N/A')}",
+                    f"Author: {result.get('author', 'N/A')}",
+                    f"Score: {result.get('score', 'N/A')}",
+                ]
+            )
+
+            if "highlights" in result and result["highlights"]:
+                formatted_results.extend(
+                    [
+                        "Highlights:",
+                        *[f"  • {highlight}" for highlight in result["highlights"]],
+                    ]
+                )
+
+            if "summary" in result and result["summary"]:
+                formatted_results.extend(["Summary:", f"  {result['summary']}"])
+
+            formatted_results.append("")
+
+        return "\n".join(formatted_results).strip()
+
+    def execute(self, input_data: ExaInputSchema, config: RunnableConfig | None = None, **kwargs) -> dict[str, Any]:
+        """
+        Executes the search using the Exa API and returns the formatted results.
+        """
+        logger.debug(f"Tool {self.name} - {self.id}: started with input data {input_data.model_dump()}")
+
+        config = ensure_config(config)
+        self.run_on_node_execute_run(config.callbacks, **kwargs)
+
+        payload = {k: v for k, v in input_data.model_dump().items() if v is not None}
+        get_contents = payload.pop("get_contents", False)
+
+        payload = {
+            "query": payload.pop("query"),
+            "useAutoprompt": payload.pop("use_autoprompt", False),
+            "type": payload.pop("type", "neural"),
+            "numResults": payload.pop("num_results", 10),
+            **{self.to_camel_case(k): v for k, v in payload.items()},
+        }
+
+        if get_contents:
+            payload["contents"] = {
```

Comment:
Let's define extra models for such `contents` data and defining it with such defaults in `ExaInputSchema`. So the user could update such params in case it would like.

Reply:
I've simplified it, due to agents stuff

Reply:
Ok, let's do now not just for Agents stuff.

Reply:
If I create a new schema for content, we won't be able to generate and use these schemas with the agents, as this could cause problems with the schema tool's description generation. 
For user behavior, it would be sufficient to add a choice: either receive additional content or not.